net/multi_port_listener_factory.go

```diff
@@ -0,0 +1,65 @@
+package net
+
+import (
+	"net"
+
+	"github.com/sirupsen/logrus"
+)
+
+type multiPortTCPListenerFactory struct {
+	addr   string
+	logger *logrus.Entry
+}
+
+type MultiPortTCPListenerFactoryArgs struct {
+	Address string
+	Logger  *logrus.Logger
+}
+
+func NewMultiPortTCPListenerFactory(args *MultiPortTCPListenerFactoryArgs) (ListenerFactory, error) {
```

Comment:
I have some nitpicks about this factory. If I'm not mistaken, since you only allow 1 address to be passed, you could only create multiple listeners if you allowed each listener to bind to a different port or have a `sockopt` set to `SO_REUSEADDR` (which I don't think you can do easily with `net.Listener`). And you do pass `:0` address in the actual server. Not sure what the based course of action is. Since we only use it in a few places, maybe we can comment here (if we can't make easy code fixes).

Reply:
Oh that's a good point. I should probably pass a host if anything. So pass in `localhost` or `0.0.0.0`. Then have the listener append `:0`